meson.build

```diff
@@ -0,0 +1,800 @@
+# TODO: cross-compile supports is rudimentary
+
+project('uftrace', 'c', version: '0.9.4')
+python_module = import('python')
+
+common_cflags = ['-D_GNU_SOURCE', '-D_DEFAULT_SOURCE', '-D_XOPEN_SOURCE=600']
+common_cflags += '-fvisibility=hidden'
+common_cflags += '-fno-omit-frame-pointer'
+common_ldflags = []
+uftrace_cflags    = []
+libmcount_cflags  = []
+demangler_cflags  = []
+symbols_cflags    = []
+dbginfo_cflags    = []
+traceevent_cflags = []
+test_cflags       = []
+
+program_shell = find_program('sh', required: true)
+program_git = find_program('git', required: true)
+
+cc = meson.get_compiler('c')
+
+if get_option('arch') == 'auto'
+    arch = host_machine.cpu_family()
+elif get_option('arch') == host_machine.cpu_family()
+    arch = host_machine.cpu_family()
+else
+    error('host compiler `@0@` doesn\'t match host architecture `@1@`'.format(
+        host_machine.cpu_family(), get_option('arch')))
```

Comment:
I'm not entirely sure what you mean. I tested on a 32-bit machine, `host_machine.cpu_family()` gives `x86` which makes the build fail because we want `i386`.

Before the above checks, I rename `x86` to `i383` (see the commits when I'll push them).